File: rb/features.rb
# GregorianLunarCalendar SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module GregorianLunarCalendarFeatures
  def self.make_feature(name)
    case name
    when "base"
      GregorianLunarCalendarBaseFeature.new
    when "test"
      GregorianLunarCalendarTestFeature.new
    else
      GregorianLunarCalendarBaseFeature.new
    end
  end
end

File: py/gregorianlunarcalendar_sdk/features.py
# GregorianLunarCalendar SDK feature factory

from gregorianlunarcalendar_sdk.feature.base_feature import GregorianLunarCalendarBaseFeature
from gregorianlunarcalendar_sdk.feature.test_feature import GregorianLunarCalendarTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GregorianLunarCalendarBaseFeature(),
        "test": lambda: GregorianLunarCalendarTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
